plugins/catalog-backend-module-bitbucket-cloud/src/BitbucketCloudEntityProvider.test.ts

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getVoidLogger } from '@backstage/backend-common';
+import { TaskInvocationDefinition, TaskRunner } from '@backstage/backend-tasks';
+import { ConfigReader } from '@backstage/config';
+import { EntityProviderConnection } from '@backstage/plugin-catalog-backend';
+import { BitbucketCloudEntityProvider } from './BitbucketCloudEntityProvider';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+
+class PersistingTaskRunner implements TaskRunner {
+  private tasks: TaskInvocationDefinition[] = [];
+
+  getTasks() {
+    return this.tasks;
+  }
+
+  run(task: TaskInvocationDefinition): Promise<void> {
+    this.tasks.push(task);
+    return Promise.resolve(undefined);
+  }
+}
+
+const logger = getVoidLogger();
+
+const server = setupServer();
+
+describe('BitbucketCloudEntityProvider', () => {
+  beforeAll(() => server.listen());
+  afterEach(() => server.resetHandlers());
+  afterEach(() => jest.resetAllMocks());
+  afterAll(() => server.close());
```

Comment:
the server ones could be replaced with the `setupRequestMockHandlers` from `@backstage/backend-test-utils` (it also ensures that unhandled calls aren't let out to leak)

Reply:
thanks for the hint. I will have a look at it.

Reply:
@freben the same seems to be provided by `@backstage/test-utils` and both are not marked as deprecated. Which one should be used?

Reply:
I see that backend-test-utils has less exports compared to test-utils

Reply:
adjusted